### ISSUE TITLE
Scaling

### ISSUE THREAD

**Issue by jodevak** [Hironsan/anago#24]
Is there a way to scale ? i tried with 12 million sentences, and its giving me that it needs 3000 hours to finish !

**Comment by Hironsan** [comment]
Use [CuDNNLSTM](https://keras.io/layers/recurrent/#cudnnlstm).

It may help you to scale.

**Comment by jodevak** [comment]
Yes, 100% that gpu based learning is way faster, but dont you think that 3000 hours is a big number ? even though its on 4 cores cpu, i still dont think this is a reasonable time.

**Comment by Hironsan** [comment]
I don't know about your dataset. So I will guess the problems.
* Plotting learning curve
* Not using CRF layer

## Plotting learning curve
I think one of the problems is the number of sentences.

Do you really need 12 million sentences?
You can check it by plotting [learning curve](http://scikit-learn.org/stable/auto_examples/model_selection/plot_learning_curve.html).

If you have not plotted it, I think you should plot it as follows:
![learning_curve.png](https://qiita-image-store.s3.amazonaws.com/0/77079/15fee0bf-d938-764f-e7c4-16e030b115c5.png)

## Not using CRF layer
I think another problem is CRF layer.

anaGo uses CRF layer as the final layer by default. But CRF increases its computation time as the number of classes is increasing. 

So if there are many classes in your dataset, you should use softmax layer or another instead of CRF layer. It improves computation time.
